safe_core/src/client/mock/vault.rs

```diff
@@ -273,41 +275,54 @@ impl Vault {
             }
             return Ok(());
         }
+
+        // Safe to unwrap as the client check happens in the previous block.
```

Comment:
perhaps we should pass the unwrapped token in here ?

Reply:
We cannot do that, as sometimes this function gets called by the client and clients send `None` for `token`. Although, we early return if it's a client/owner in the previous block above this line, making it safe to unwrap.